modules/monitoring-configuring-pod-topology-spread-constraints.adoc

```diff
@@ -0,0 +1,189 @@
+// Module included in the following assemblies:
+//
+// * observability/monitoring/configuring-the-monitoring-stack.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="configuring-pod-topology-spread-constraints_{context}"]
+= Configuring pod topology spread constraints
+
+You can configure pod topology spread constraints for 
+ifndef::openshift-dedicated,openshift-rosa[]
+the pods deployed by the Cluster Monitoring Operator
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+the pods for user-defined monitoring
+endif::openshift-dedicated,openshift-rosa[]
+to control how pod replicas are scheduled to nodes across zones.
+This ensures that the pods are highly available and run more efficiently, because workloads are spread across nodes in different data centers or hierarchical infrastructure zones.
+
+You can configure pod topology spread constraints for monitoring pods by using 
+ifndef::openshift-dedicated,openshift-rosa[]
+the `cluster-monitoring-config` or 
+endif::openshift-dedicated,openshift-rosa[]
+the `user-workload-monitoring-config` config map.
+
+.Prerequisites
+
+ifndef::openshift-dedicated,openshift-rosa[]
+* *If you are configuring pods for core {product-title} monitoring:*
+** You have access to the cluster as a user with the `cluster-admin` cluster role.
+** You have created the `cluster-monitoring-config` `ConfigMap` object.
+* *If you are configuring pods for user-defined monitoring:*
+** A cluster administrator has enabled monitoring for user-defined projects.
+** You have access to the cluster as a user with the `cluster-admin` cluster role, or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+** You have created the `user-workload-monitoring-config` `ConfigMap` object.
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `dedicated-admin` role.
+* The `user-workload-monitoring-config` `ConfigMap` object exists. This object is created by default when the cluster is created.
+endif::openshift-dedicated,openshift-rosa[]
+
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+ifndef::openshift-dedicated,openshift-rosa[]
+* *To configure pod topology spread constraints for core {product-title} monitoring:*
+
+. Edit the `cluster-monitoring-config` config map in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add the following settings under the `data/config.yaml` field to configure pod topology spread constraints:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    <component>: # <1>
+      topologySpreadConstraints:
+      - maxSkew: <n> # <2>
+        topologyKey: monitoring # <3>
```

Comment:
maybe
```suggestion
        topologyKey: <key> # <3>
```
use can set value for topologyKey, it is not a fixed value

Reply:
Oh, yes, but does it have to be `monitoring` in this case? Because I am a little bit hesitant pointing that value out, if it should always be set to `monitoring` for this case :sweat_smile:

Reply:
no, user can set any value based on their needs. see explanation in https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints. the example in this PR used `topologyKey: monitoring`, it's fine, no need to change
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
    prometheusK8s:
      topologySpreadConstraints:
      - maxSkew: 1
        topologyKey: monitoring
        whenUnsatisfiable: DoNotSchedule
        labelSelector:
          matchLabels:
            app.kubernetes.io/name: prometheus
```